### ISSUE TITLE
Needs a requirements file

### ISSUE THREAD

**Issue by rohun-tripathi** [bcdnlp/FAITHSCORE#4]
Ran into multiple issues installing in a python 3.11 conda enviroment.
A pip freeze of the environment in which it worked will be very useful

**Comment by amith-ananthram** [comment]
i compared the date in the gitblame for the README against the releases of the various packages it relies on and came up with this list (was able to get it working after some trial & error):

```
absl-py==2.2.1
accelerate==0.21.0
addict==2.4.0
aiofiles==23.2.1
aiohappyeyeballs==2.6.1
aiohttp==3.11.12
aiosignal==1.3.2
aliyun-python-sdk-core==2.16.0
aliyun-python-sdk-kms==2.16.5
altair==5.5.0
annotated-types==0.7.0
antlr4-python3-runtime==4.9.3
anyio==4.9.0
async-timeout==5.0.1
attrs==25.3.0
audioread==3.0.1
bitarray==2.8.3
bitsandbytes==0.41.0
certifi==2025.1.31
cffi==1.17.1
charset-normalizer==3.4.1
click==8.1.8
cloudpickle==3.1.1
cmake==4.0.0
colorama==0.4.6
contourpy==1.3.1
crcmod==1.7
cryptography==44.0.2
cycler==0.12.1
datasets==2.13.0
decorator==5.2.1
decord==0.6.0
diffusers==0.32.2
dill==0.3.6
einops==0.6.1
einops-exts==0.0.4
exceptiongroup==1.2.2
fairseq==0.12.2
fastapi==0.115.12
ffmpy==0.5.0
filelock==3.18.0
fonttools==4.56.0
frozenlist==1.5.0
fsspec==2024.9.0
ftfy==6.3.1
gast==0.6.0
gradio==3.35.2
gradio_client==0.2.9
grpcio==1.71.0
h11==0.14.0
httpcore==0.17.3
httpx==0.24.0
huggingface-hub==0.17.3
hydra-core==1.3.2
idna==3.10
importlib_metadata==8.6.1
importlib_resources==6.5.2
Jinja2==3.1.6
jmespath==0.10.0
joblib==1.4.2
jsonschema==4.23.0
jsonschema-specifications==2024.10.1
kiwisolver==1.4.8
latex2mathml==3.77.0
lazy_loader==0.4
librosa==0.10.1
lightning-utilities==0.14.2
linkify-it-py==2.0.3
lit==18.1.8
-e git+https://github.com/haotian-liu/LLaVA.git@786aa6a19ea10edc6f574ad2e16276974e9aaa3a#egg=llava
llvmlite==0.44.0
lxml==5.3.1
Markdown==3.7
markdown-it-py==2.2.0
markdown2==2.5.3
MarkupSafe==2.1.5
matplotlib==3.10.1
mdit-py-plugins==0.3.3
mdurl==0.1.2
modelscope==1.9.5
mpmath==1.3.0
msgpack==1.1.0
multidict==6.3.0
multiprocess==0.70.14
narwhals==1.33.0
networkx==3.4.2
nltk==3.9.1
numba==0.61.0
numpy==1.26.4
nvidia-cublas-cu11==11.10.3.66
nvidia-cublas-cu12==12.1.3.1
nvidia-cuda-cupti-cu11==11.7.101
nvidia-cuda-cupti-cu12==12.1.105
nvidia-cuda-nvrtc-cu11==11.7.99
nvidia-cuda-nvrtc-cu12==12.1.105
nvidia-cuda-runtime-cu11==11.7.99
nvidia-cuda-runtime-cu12==12.1.105
nvidia-cudnn-cu11==8.5.0.96
nvidia-cudnn-cu12==8.9.2.26
nvidia-cufft-cu11==10.9.0.58
nvidia-cufft-cu12==11.0.2.54
nvidia-curand-cu11==10.2.10.91
nvidia-curand-cu12==10.3.2.106
nvidia-cusolver-cu11==11.4.0.1
nvidia-cusolver-cu12==11.4.5.107
nvidia-cusparse-cu11==11.7.4.91
nvidia-cusparse-cu12==12.1.0.106
nvidia-nccl-cu11==2.14.3
nvidia-nccl-cu12==2.18.1
nvidia-nvjitlink-cu12==12.8.93
nvidia-nvtx-cu11==11.7.91
nvidia-nvtx-cu12==12.1.105
omegaconf==2.3.0
openai==0.27.0
opencv-python==4.11.0.86
orjson==3.10.16
oss2==2.19.1
packaging==24.2
pandas==2.2.3
peft==0.4.0
pillow==10.4.0
platformdirs==4.3.7
pooch==1.8.2
portalocker==3.1.1
propcache==0.3.1
protobuf==3.20.1
psutil==7.0.0
pyarrow==19.0.1
pycocoevalcap==1.2
pycocotools==2.0.8
pycparser==2.22
pycryptodome==3.22.0
pydantic==1.10.21
pydantic_core==2.33.0
pyDeprecate==0.3.2
pydot==3.0.4
pydub==0.25.1
Pygments==2.19.1
pyparsing==3.2.3
python-dateutil==2.9.0.post0
python-multipart==0.0.20
pytorch-lightning==1.6.5.post0
pytz==2025.2
PyYAML==6.0.2
RapidFuzz==3.12.2
referencing==0.36.2
regex==2024.11.6
requests==2.32.3
rich==14.0.0
rouge-score==0.0.4
rpds-py==0.24.0
ruff==0.11.2
sacrebleu==2.5.1
safetensors==0.5.3
scikit-learn==1.2.2
scipy==1.15.2
semantic-version==2.10.0
sentencepiece==0.1.99
shellingham==1.5.4
shortuuid==1.0.13
simplejson==3.20.1
six==1.17.0
sniffio==1.3.1
sortedcontainers==2.4.0
soundfile==0.13.1
soxr==0.5.0.post1
starlette==0.46.1
svgwrite==1.4.3
sympy==1.13.3
tabulate==0.9.0
tensorboard==2.19.0
tensorboard-data-server==0.7.2
threadpoolctl==3.6.0
timm==0.6.13
tokenizers==0.13.3
tomli==2.2.1
tomlkit==0.12.0
torch==2.1.0
torchaudio==2.1.0
torchmetrics==1.7.0
torchvision==0.15.2
tqdm==4.67.1
transformers==4.31.0
triton==2.1.0
typer==0.15.2
typing-inspection==0.4.0
typing_extensions==4.13.0
tzdata==2025.2
uc-micro-py==1.0.3
unicodedata2==16.0.0
urllib3==2.3.0
uvicorn==0.34.0
wavedrom==2.0.3.post3
wcwidth==0.2.13
websockets==11.0.3
Werkzeug==3.1.3
xxhash==3.5.0
yapf==0.43.0
yarl==1.18.3
zhconv==1.4.3
zipp==3.21.0
``
